tests/check_new_syntax.py

```diff
@@ -26,7 +26,23 @@ def visit_If(self, node: ast.If) -> None:
                 )
             self.generic_visit(node)
 
+    class PEP570Finder(ast.NodeVisitor):
+        def _visit_function(self, node: ast.FunctionDef | ast.AsyncFunctionDef) -> None:
+            self.lineno = node.lineno
+            self.generic_visit(node)
+
+        visit_FunctionDef = visit_AsyncFunctionDef = _visit_function
```

Comment:
This dance is necessary because `ast.arguments` instances don't have `lineno` attributes